include/robot_interfaces/cereal_eigen.hpp

```diff
@@ -0,0 +1,46 @@
+/**
+ * @file
+ * @brief Serialization functions for serializing Eigen matrices and arrays
+ *        with cereal.
+ * @authors Azoth, eudoxos
+ * @date 2020-01-15
+ * @license CC BY-SA 4.0
+ * @todo Move this to some "serialization tools" package.
+ *
+ * Taken from https://stackoverflow.com/a/51944389/2095383.
+ */
+#include <Eigen/Eigen>
+#include <type_traits>
+
+namespace cereal
+{
+template <class Archive, class Derived>
+inline typename std::enable_if<
+    traits::is_output_serializable<BinaryData<typename Derived::Scalar>,
+                                   Archive>::value,
+    void>::type
+save(Archive& ar, Eigen::PlainObjectBase<Derived> const& m)
+{
+    typedef Eigen::PlainObjectBase<Derived> ArrT;
+    if (ArrT::RowsAtCompileTime == Eigen::Dynamic) ar(m.rows());
```

Comment:
i don't understand what this does

Reply:
It only adds the size of the matrix/array to the serialized data if it is a dynamic-size type and thus this information is needed to deserialise.  I.e. for something like a Vector3d the size is omitted as it is anyway known.